bloomfilter/src/main/kotlin/org/kethereum/bloomfilter/BloomFilter.kt

```diff
@@ -0,0 +1,37 @@
+package org.kethereum.bloomfilter
+
+import org.kethereum.hashes.sha256
+import java.math.BigInteger
+import java.util.BitSet
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+
+class BloomFilter(private val size: Int) {
+    private val bits = BitSet(size)
+    private val lock = ReentrantReadWriteLock()
+
+    fun add(value: ByteArray) {
+        lock.write {
+            for (seed in 1..3) {
+                bits.set(hashing(size, seed, value))
+            }
+        }
+    }
+
+    fun mightContains(value: ByteArray): Boolean {
+        lock.read {
+            for (seed in 1..3) {
+                if (!bits.get(hashing(size, seed, value))) {
+                    return false
+                }
+            }
+        }
+        return true
+    }
+
+    private fun hashing(filterSize: Int, seed: Int, value: ByteArray) =
+            BigInteger(1, value.plus(seed.toByte()).sha256())
```

Comment:
Using [sha256](https://github.com/komputing/KEthereum/tree/master/hashes/src/main/kotlin/org/kethereum/hashes) which is under `hashes` package for now but the hash algorithm may not appropriate. I'd like to try to implement more lightweight algorithm into the `hashes` package. 👀